src/eos-updater-poll-common.c

```diff
@@ -372,32 +427,27 @@ fetch_latest_commit (OstreeRepo *repo,
                                  error))
     return FALSE;
 
-  /* If this is a redirect commit, follow it and fetch the new ref instead */
+  /* If this is a redirect commit, follow it and fetch the new ref instead
+   * (unless the rebase is a loop; ignore that). */
   metadata = g_variant_get_child_value (commit, 0);
   rebase = g_variant_lookup_value (metadata, "ostree.endoflife-rebase", G_VARIANT_TYPE_STRING);
 
-  if (rebase != NULL)
+  if (rebase != NULL &&
+      g_strcmp0 (g_variant_get_string (rebase, NULL), ref) != 0)
     {
       g_clear_pointer (&ref, g_free);
       ref = g_variant_dup_string (rebase, NULL);
 
-      g_clear_pointer (&options, g_variant_unref);
-      options = get_repo_pull_options (url_override, ref);
-      if (!ostree_repo_pull_with_options (repo,
-                                          remote_name,
-                                          options,
-                                          NULL,
-                                          cancellable,
-                                          error))
-        return FALSE;
-
       g_clear_pointer (&checksum, g_free);
```

Comment:
Hmm, clearing the checksum outparam here as a way of indicating that we hit a rebase feels like a non-intuitive design to me - perhaps we could have a function that just returns the ref to "follow" if we need to follow a new one and then another one which actually fetches the checksum.

Reply:
Yeah, it would be nice to rework this a bit. I’d prefer to do that as a follow-up commit, since I was just extending the existing behaviour rather than introducing new behaviour here.

Reply:
Pushed a follow-up commit.